.env.example

```diff
@@ -1,24 +1,16 @@
-## Example .env file for the development environment
-## Rename this file to .env before starting the server
-SECRET_KEY= # type your secret key here
 DEBUG=True
 ALLOWED_HOSTS=.localhost, 127.0.0.1
 TIME_ZONE=America/Chicago
 DJANGO_LOG_LEVEL=WARN
 DJANGO_DB_LOG_LEVEL=DEBUG
 DSSG_LOG_LEVEL=DEBUG
 
+# SQLite DB
+DATABASE_URL=sqlite:///db.sqlite3
+# Or for PostgreSQL
+# DATABASE_URL=postgres://USER:PASSWORD@HOST:PORT/DBNAME
 
-EMAIL_HOST=localhost
-EMAIL_PORT=587
-EMAIL_HOST_USER=myuser
-EMAIL_HOST_PASSWORD=mypassword
-EMAIL_USE_TLS=True
-EMAIL_FROM_ADDRESS=info@dssg.com
-
-SITE_NAME=DSSG Solve
+# Optional overrides
+# SITE_NAME=DSSG Solve
 
-## SQLite DB
-DATABASE_URL=sqlite:///db.sqlite3
-## Or for PostgreSQL
-#DATABASE_URL=postgres://USER:PASSWORD@HOST:PORT/DBNAME
+# Programmatically generated
```

Comment:
There are two things going on here:

* Rather than ask a developer to come up with a good secret key, (and for that matter require that they follow set-up steps in the documentation), `develop` just makes them a `.env` file, populated with a randomly-generated `SECRET_KEY`

* We're not shipping a `.env` file with the Docker image or anything like that, and there was no time to work out a sophisticated solution for managing and loading configuration and/or secrets, (which would be say stored encrypted in S3 and loaded by the Docker container upon initialization). Rather, I specified necessary configuration via the process environment, in configuration of the "task" (*i.e.* configuration of the Docker container run); (this is far from a terrible hack, especially considering we're not yet specifying database credentials this way). However, I had to fight a good deal with the integration with `decouple` – particularly any variable lacking a default value, and given a dummy value in `.env`. It seems unwise to me to require the specification of these dummy values in production, and misleading to do so even in development. It's great to _document_ these options, "in-code," by having them in `.env.example`; but, where unnecessary, they should be commented out. And, as for the email-related settings, Django already has reasonable defaults, (even in some cases `None`), which are preferable to `myuser` and `mypassword`; so, in the case of these, I even removed the (currently unnecessary) specification of these settings from the `settings.py` module (see below).

Reply:
I feel like I'm missing something, regarding the email configuration settings that were commented out from the settings file/env file. When deploying the site in production you need to set your SMTP settings (host, port, credentials) for whatever email service you may be using, and you don't want the from address to be 'webmaster@localhost.com' (which is the Django default). AFAIK the only way of specifying those values is through the settings.py file; what's the plan for setting them when you deploy with DEBUG=False and the server tries to send emails for real? Are you going to include them back in settings.py, or are you going to pass the values for those parameters in a different way?

Reply:
Yes, of course these will need to be specified for SMTP (or whatever mail server protocol we use – probably the SMTP relay to the AWS mail service API).

However, there is no SMTP server right now, and I don't think it was worth consideration for the demo. But, specification of these settings from `config(...)`, without defaults, required that I set dummy environment variables in production, like `EMAIL_HOST_USER=myuser`, which didn't make sense to me.

…Especially considering that these settings aren't, as far as I could tell, used in development mode either, (which doesn't pretend to use SMTP).

So, I commented these out. When integrating with an SMTP server, these will need to be set, (though I'd recommend not _requiring_ their specification outside of an environment which can actually use them).